sdc/tests/tests_perf/test_perf_series_str.py

```diff
@@ -135,42 +73,28 @@ def _test_series_str(self, pyfunc, name, input_data=None):
 
             exec_times, boxing_times = get_times(hpat_func, test_data, iter_number=self.iter_number)
-            self.test_results.add(name, 'JIT', test_data.size, exec_times, data_width,
+            self.test_results.add(test_name, 'SDC', test_data.size, exec_times, data_width,
                                   boxing_times, compile_results=compile_results, num_threads=self.num_threads)
             exec_times, _ = get_times(pyfunc, test_data, iter_number=self.iter_number)
-            self.test_results.add(name, 'Reference', test_data.size, exec_times, data_width,
+            self.test_results.add(test_name, 'Python', test_data.size, exec_times, data_width,
                                   num_threads=self.num_threads)
 
-    def test_series_str_len(self):
-        self._test_series_str(usecase_series_len, 'series_str_len')
-
-    def test_series_str_capitalize(self):
-        self._test_series_str(usecase_series_capitalize, 'series_str_capitalize')
-
-    def test_series_str_lower(self):
-        self._test_series_str(usecase_series_lower, 'series_str_lower')
-
-    def test_series_str_swapcase(self):
-        self._test_series_str(usecase_series_swapcase, 'series_str_swapcase')
-
-    def test_series_str_title(self):
-        self._test_series_str(usecase_series_title, 'series_str_title')
-
-    def test_series_str_upper(self):
-        self._test_series_str(usecase_series_upper, 'series_str_upper')
-
-    def test_series_str_lstrip(self):
-        input_data = ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]
-        self._test_series_str(usecase_series_lstrip, 'series_str_lstrip', input_data=input_data)
-
-    def test_series_str_rstrip(self):
-        input_data = ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]
-        self._test_series_str(usecase_series_rstrip, 'series_str_rstrip', input_data=input_data)
-
-    def test_series_str_strip(self):
-        input_data = ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]
-        self._test_series_str(usecase_series_strip, 'series_str_strip', input_data=input_data)
-
 
-if __name__ == "__main__":
-    unittest.main()
+# (method_name, parameters, input_data)
+cases = [
+    ('center', '1', test_global_input_data_unicode_kind1),
+    ('endswith', '"e"'),
+    ('find', '"e"'),
+    ('len', ''),
+    ('ljust', '1', test_global_input_data_unicode_kind1),
+    ('lower', ''),
+    ('lstrip', '', ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]),
+    ('rjust', '1', test_global_input_data_unicode_kind1),
+    ('rstrip', '', ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]),
+    ('startswith', '"e"'),
+    ('strip', '', ['\t{}  '.format(case) for case in test_global_input_data_unicode_kind4]),
+    ('upper', ''),
+    ('zfill', '1', test_global_input_data_unicode_kind1),
+]
+
+generate_test_cases(cases, TestSeriesStringMethods, 'series', 'str.')
```

Comment:
I propose use prefix without dot `'str'`, then add the dot inside of the `test_gen`.

Reply:
No, this cannot be, because then in cases without a prefix we will have a call with two points in uscase, for example, input..min ().

Reply:
`'.'.join(...)` can help you.